sycl/test/aie/metaprogramming.cpp

```diff
@@ -0,0 +1,88 @@
+// REQUIRES: aie
+
+// RUN: %aie_clang %s -o %t.bin
+
```

Comment:
I guess that most of thi stuff is not for the end-user?
```suggestion
// Exercise some implementation details
```